python/ray/train/lightning/_lightning_utils.py

```diff
@@ -65,6 +77,25 @@ def distributed_sampler_kwargs(self) -> Dict[str, Any]:
             rank=self.global_rank,
         )
 
+    def lightning_module_state_dict(self) -> Dict[str, Any]:
+        """Gathers the full state dict to rank 0 on CPU."""
+        assert self.model is not None, "Failed to get the state dict for a None model!"
+
+        if _LIGHTNING_GREATER_EQUAL_2_0 and _TORCH_FSDP_AVAILABLE:
+            with FullyShardedDataParallel.state_dict_type(
```

Comment:
should we add a TODO? this is one of the approaches to get the full model right?
if the distributed checkpointing is in, would be nice if the user can choose which approach to use?

Reply:
yes I totally agree. This is not the most efficient way to save checkpoint for large model. It'd be super helpful if we support distributed checkpoint.